source/common/upstream/load_balancer_impl.cc

```diff
@@ -81,18 +81,23 @@ void LoadBalancerBase::recalculatePerPriorityState(uint32_t priority) {
     per_priority_load_[0] = 100;
     return;
   }
+
   size_t total_load = 100;
+  int32_t first_healthy_priority = -1;
   for (size_t i = 0; i < per_priority_health_.size(); ++i) {
-    // Now assign as much load as possible to the high priority levels and cease assigning load when
-    // total_load runs out.
+    if (first_healthy_priority < 0 && per_priority_health_[i]) {
+      first_healthy_priority = i;
+    }
+    // Now assign as much load as possible to the high priority levels and cease assigning load
+    // when total_load runs out.
     per_priority_load_[i] =
         std::min<uint32_t>(total_load, per_priority_health_[i] * 100 / total_health);
     total_load -= per_priority_load_[i];
   }
+
   if (total_load != 0) {
-    // Account for rounding errors.
```

Comment:
Can we keep this comment? And probably add more with your explanation in this PR desc. I think that will be useful for future readers.

Reply:
Ah yeah I wasn't intentionally trying to remove it. Will add back